### PR TITLE
041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omit…

### PR DESCRIPTION
**Pull request by istoimenova**
Script is part of TS: #932

### CODE REVIEW

**Comment by istoimenova** [comment]
@OVikhrov, @dboltovskyi, please review.

File: test_scripts/API/SetAudioStreamingIndicator/041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omitted_PTgroups.lua [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+local function SetAudioStreamingIndicator_omit_Base4()
+  commonPreconditions:BackupFile("sdl_preloaded_pt.json")
+
+  local pathToFile = config.pathToSDL .. 'sdl_preloaded_pt.json'
+  local file = io.open(pathToFile, "r")
+  local json_data = file:read("*all")
+  file:close()
+
+  local data_preloaded = json.decode(json_data)
+  if(data.policy_table.functional_groupings["DataConsent-2"]) then
```
Comment (dboltovskyi) [inline test_scripts/API/SetAudioStreamingIndicator/041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omitted_PTgroups.lua]: @istoimenova , 'data' is undefined variable

File: test_scripts/API/SetAudioStreamingIndicator/041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omitted_PTgroups.lua [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+--[[ Local Functions ]]
+
+--[[@SetAudioStreamingIndicator_omit_Base4: update preloaded_pt.json
+! SetAudioStreamingIndicator is not included in Base-4 functional group
+! @parameters: NO
+]]
+local function SetAudioStreamingIndicator_omit_Base4()
+  commonPreconditions:BackupFile("sdl_preloaded_pt.json")
+
+  local pathToFile = config.pathToSDL .. 'sdl_preloaded_pt.json'
```
Comment (dboltovskyi) [inline test_scripts/API/SetAudioStreamingIndicator/041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omitted_PTgroups.lua]: @istoimenova , 'config.pathToSDL' parameter doesn't end up with "/" by default, therefore 'pathToFile' will be incorrect

File: test_scripts/API/SetAudioStreamingIndicator/041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omitted_PTgroups.lua [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  local file = io.open(pathToFile, "r")
+  local json_data = file:read("*all")
+  file:close()
+
+  local data_preloaded = json.decode(json_data)
+  if(data.policy_table.functional_groupings["DataConsent-2"]) then
+    data.policy_table.functional_groupings["DataConsent-2"].rpcs = { json.null }
+  end
+  data_preloaded.policy_table.functional_groupings["Base-4"].rpcs.SetAudioStreamingIndicator = json.null
```
Comment (dboltovskyi) [inline test_scripts/API/SetAudioStreamingIndicator/041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omitted_PTgroups.lua]: @istoimenova , 'Base-4' is a only one default group.
If it's removed, there will be no groups assigned to application when it's registered
Anyway why we remove this group? "SetAudioStreamingIndicator" RPC is not included in it.

File: test_scripts/API/SetAudioStreamingIndicator/041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omitted_PTgroups.lua [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+commonSteps:DeletePolicyTable()
+
+--[[ General Settings for configuration ]]
+Test = require('connecttest')
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:Precondition_ActivateApp()
```
Comment (dboltovskyi) [inline test_scripts/API/SetAudioStreamingIndicator/041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omitted_PTgroups.lua]: @istoimenova , If script is designed for PROPRIETARY flow activation is not required. Default permissions will be assigned once application registered

File: test_scripts/API/SetAudioStreamingIndicator/041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omitted_PTgroups.lua [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  file:close()
+
+  local data_preloaded = json.decode(json_data)
+  if(data.policy_table.functional_groupings["DataConsent-2"]) then
+    data.policy_table.functional_groupings["DataConsent-2"].rpcs = { json.null }
+  end
+  data_preloaded.policy_table.functional_groupings["Base-4"].rpcs.SetAudioStreamingIndicator = json.null
+  
+  data_preloaded = json.encode(data_preloaded)
+  file = io.open(config.pathToSDL .. 'sdl_preloaded_pt.json', "w")
```
Comment (dboltovskyi) [inline test_scripts/API/SetAudioStreamingIndicator/041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omitted_PTgroups.lua]: @istoimenova , 'config.pathToSDL' parameter doesn't end up with "/" by default, therefore 'file' descriptor will be incorrect

File: test_scripts/API/SetAudioStreamingIndicator/041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omitted_PTgroups.lua [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+commonSteps:DeleteLogsFiles()
+commonSteps:DeletePolicyTable()
+
+--[[ General Settings for configuration ]]
+Test = require('connecttest')
+require('user_modules/AppTypes')
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
```
Comment (OVikhrov) [inline test_scripts/API/SetAudioStreamingIndicator/041_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_media_app_RPC_omitted_PTgroups.lua]: @istoimenova Please activate and consent device and then send request. Currently you get DISALLOWED becouse  device not consented

**Pull request merged**